contracts/Bridge.sol

```diff
@@ -234,57 +250,50 @@ contract Bridge {
   // netVersion is for EIP155 - v = netVersion*2 + 35 or netVersion*2 + 36
   // This can be found in a web3 console with web3.version.network. Parity
   // also serves it in the transaction log under `chainId`
-  function prepWithdraw(bytes nonce, bytes gasPrice, bytes gasLimit, bytes v,
-  bytes r, bytes s, address[3] addrs, uint256 amount, bytes32 txRoot, bytes path,
-  bytes parentNodes, bytes netVersion) public {
+
+  function prepWithdraw(
+    bytes v,
+    bytes32[3] b32p, // r=0, s=1, txRoot=2
+    //bytes32 r, 
+    //bytes32 s, 
+    address[3] addrs, 
+    uint256 amount, 
+    //bytes32 txRoot, 
+    bytes path,
+    bytes parentNodes, 
+    bytes netVersion,
+    bytes rlpDepositTx,
+    bytes rlpWithdrawTx
+    )
+    public 
+    {
+
+    // LAZ:
+    // further reduction of parameters: find method to fetch values directly
+    // from the rlp-bytes. Reduction of this function is possible, but it cannot
+    // be merged with proveReceipt, see comments there.   
+
     //assert(tokens[addrs[0]][addrs[3]] == addrs[1]);
     // Form the transaction data.
-    bytes[] memory rawTx = new bytes[](9);
-    rawTx[0] = nonce;
-    rawTx[1] = gasPrice;
-    rawTx[2] = gasLimit;
-    rawTx[3] = toBytes(addrs[0]);
-    // Leave msg.value blank. This means only token-token transfers for now.
-    rawTx[4] = hex"";
-    //8340f549 function signature of "deposit(address,address,uint256)"
-    rawTx[5] = BytesLib.concat(hex"8340f549",
-      BytesLib.concat(encodeAddress(addrs[1]),
-      BytesLib.concat(encodeAddress(address(this)),
-      toBytes(amount)
-    )));
-    rawTx[6] = v;
-    rawTx[7] = r;
-    rawTx[8] = s;
-    bytes memory tx = RLPEncode.encodeList(rawTx);
 
     // Make sure this transaction is the value on the path via a MerklePatricia proof
-    assert(MerklePatriciaProof.verify(tx, path, parentNodes, txRoot) == true);
+    assert(MerklePatriciaProof.verify(rlpDepositTx, path, parentNodes, b32p[2]) == true);
```

Comment:
First thing is: I essentially refactored, and thus I did not apply any functional changes. Therefor, if there is a redundancy in the code, I most possibly have adopted it.

I don't have a perfect understanding (as I had to avoid as much as possible to understand), but here is my (possibly false) understanding:

The first tx is the recreated "deposit tx already made", the second tx is the "prepared withdrawal tx":

In the original (functionally identical) code, the same memory parameter is used (tx) to hold those 2 transactions (which are very similar):
https://github.com/lazaridiscom/cryptobridge-contracts/commit/99a400a6768a1ebe558d214e312fde33e48273a8#diff-71814336e4588f369ee6cb36a9471af1L242

You notice that the original 1st tx is just slightly modified to for the final 2nd tx (v-r-s modification)

As I had some problems i directly modifying the rlpDepositTx array, I went on to pass in 2 arrays.

(I just now realize that its most possibly the same transaction, just in another form, e.g. 2 standards. If so, a simply rename would do it)